windows/bleachbit.nsi

```diff
@@ -17,163 +17,731 @@
+; !insertmacro MUI_LANGUAGE "SPANISHINTERNATIONAL"
   !insertmacro MUI_LANGUAGE "Swedish"
+; !insertmacro MUI_LANGUAGE "TATAR"
   !insertmacro MUI_LANGUAGE "Thai"
   !insertmacro MUI_LANGUAGE "TradChinese"
   !insertmacro MUI_LANGUAGE "Turkish"
   !insertmacro MUI_LANGUAGE "Ukrainian"
   !insertmacro MUI_LANGUAGE "Uzbek"
   !insertmacro MUI_LANGUAGE "Vietnamese"
+; !insertmacro MUI_LANGUAGE "WELSH"
 !endif
 
-
 !include NsisMultiUserLang.nsh
 
+!include bleachbit_lang.nsh
+
+
 ;--------------------------------
-;Function
+;License used in MUI_PAGE_LICENSE
+
+; For maybe later... Tobias.
+; LangString BLEACHBIT_LICENSE ${LANG_ENGLISH} "..\COPYING"
+; LangString BLEACHBIT_LICENSE ${LANG_FRENCH} "..\COPYING_Fre"
+; LangString BLEACHBIT_LICENSE ${LANG_GERMAN} "..\COPYING_Ger"
+
+
+;--------------------------------
+;PageEx License
+
+; For maybe later... Tobias.
+
+; Translations of "License"
+; LangString License_Text ${LANG_ENGLISH} "License"
+; LangString License_Text ${LANG_FRENCH} "Licence"
+; LangString License_Text ${LANG_GERMAN} "Lizenz"
+
+; LicenseLangString
+; Does the same as LangString only it loads the string from a text/RTF file and defines
+; a special LangString that can be used only by LicenseData.
+; LicenseLangString License_Data ${LANG_ENGLISH} license-english.txt
+; LicenseLangString License_Data ${LANG_FRENCH} license-french.txt
+; LicenseLangString License_Data ${LANG_GERMAN} license-german.txt
+
+; LicenseData
+; Specifies a text file or a RTF file to use for the license that the user can read.
+; Omit this to not have a license displayed. Note that the file must be in the evil DOS text format (\r\n, yeah!).
+; If you make your license file a RTF file it is recommended you edit it with WordPad and not MS Word.
+; Using WordPad will result in a much smaller file.
+
+; https://nsis.sourceforge.io/Reference/PageEx
+; PageEx License
+  ; LicenseText $(License_Text)
+  ; LicenseData $(License_Data)
+  ; LicenseForceSelection checkbox
+; PageExEnd
+
+
+;--------------------------------
+;Macro/Function "Uninstall"
+
+; We need to move the code from Section "Uninstall" into a function that he can
+; be executed from command line with /uninstall, too.
+; As we only can call functions starting with "un." from the uninstaller section,
+; and only functions without "un." from the installer section, we have to move the
+; funtion into a macro.
+
+!macro uninstallmacro un
+  Function ${un}uninstallfunction
+    ; Core:
+    RMDir /r "$INSTDIR"
+    DeleteRegKey HKCU "Software\${prodname}"
+
+    ; Delete normal, Start menu shortcuts
+    RMDir /r "$SMPROGRAMS\${prodname}"
+
+    ; Delete Desktop shortcut
+    Delete "$DESKTOP\BleachBit.lnk"
+
+    ; Delete Quick launch shortcut
+    Delete "$QUICKLAUNCH\BleachBit.lnk"
+
+    ; Delete Autostart shortcut
+    Delete "$SMSTARTUP\BleachBit.lnk"
+
+    ; Remove file association (Shredder)
+    DeleteRegKey HKCR "AllFileSystemObjects\shell\shred.bleachbit"
```

Comment:
I see a huge amount of code changes, and I'm not sure what the purpose of all them is. Please see my comments about splitting up the PR and commit messages

Reply:
I also resorted the code (the functions) for better (logical) readability... But GitHub don't "understand" this... So e.g. this lines are still (unchanged) code from you! It have just a other position in the document...

Reply:
Addition: If you have now a look at my files at a whole, it should have saved use many time!
I re-wrote many things and re-structured them, so you have to watch the changes now just ones...
...but I will in future make better commit messages! I promise! ^^
But not yet, to many changes, I add them on the bottom (with screenshots they show) as comment... ^^